pkg/apis/garden/v1beta1/conversions.go

```diff
@@ -1272,3 +1268,1571 @@ func volumeTypesHaveName(volumeTypes []garden.VolumeType, name string) bool {
+				Kind:       "InfrastructureConfig",
+			},
+		}
+		var infrastructureConfigOrigin string
+		if out.Spec.Provider.InfrastructureConfig != nil {
+			switch {
+			case out.Spec.Provider.InfrastructureConfig.Object != nil:
+				infrastructureConfigOrigin = "object"
+				var ok bool
+				infrastructureConfig, ok = out.Spec.Provider.InfrastructureConfig.Object.(*awsv1alpha1.InfrastructureConfig)
+				if !ok {
+					klog.Errorf("Cannot cast infrastructureConfig of garden.sapcloud.io.Shoot %s", in.Name)
+				}
+			case out.Spec.Provider.InfrastructureConfig.Raw != nil:
+				infrastructureConfigOrigin = "raw"
+				if _, _, err := decoder.Decode(out.Spec.Provider.InfrastructureConfig.Raw, nil, infrastructureConfig); err != nil {
+					// If an error occurs then the provider config information contains invalid syntax, and in this
+					// case we don't want to fail here. We rather don't try to migrate.
+					klog.Errorf("Cannot decode infrastructureConfig of garden.sapcloud.io.Shoot %s", in.Name)
+				}
+			}
+		}
+
+		if len(in.Spec.Cloud.AWS.Networks.Internal) != len(in.Spec.Cloud.AWS.Zones) {
+			return fmt.Errorf("aws internal networks must have same number of entries like zones")
+		}
+		if len(in.Spec.Cloud.AWS.Networks.Public) != len(in.Spec.Cloud.AWS.Zones) {
+			return fmt.Errorf("aws public networks must have same number of entries like zones")
+		}
+		if len(in.Spec.Cloud.AWS.Networks.Workers) != len(in.Spec.Cloud.AWS.Zones) {
+			return fmt.Errorf("aws workers networks must have same number of entries like zones")
+		}
+
+		zones := make([]awsv1alpha1.Zone, 0, len(in.Spec.Cloud.AWS.Zones))
+		for i, zone := range in.Spec.Cloud.AWS.Zones {
+			zones = append(zones, awsv1alpha1.Zone{
+				Name:     zone,
+				Internal: in.Spec.Cloud.AWS.Networks.Internal[i],
+				Public:   in.Spec.Cloud.AWS.Networks.Public[i],
+				Workers:  in.Spec.Cloud.AWS.Networks.Workers[i],
+			})
+		}
+
+		var vpcCIDR *string
+		if c := in.Spec.Cloud.AWS.Networks.VPC.CIDR; c != nil {
+			cidr := string(*c)
+			vpcCIDR = &cidr
+		}
+
+		infrastructureConfig.Networks = awsv1alpha1.Networks{
+			VPC: awsv1alpha1.VPC{
+				ID:   in.Spec.Cloud.AWS.Networks.VPC.ID,
+				CIDR: vpcCIDR,
+			},
+			Zones: zones,
+		}
+
+		var infrastructureConfigRawExtension runtime.RawExtension
+		if infrastructureConfigOrigin == "object" {
+			infrastructureConfigRawExtension.Object = infrastructureConfig
+		} else {
+			data, err := json.Marshal(infrastructureConfig)
+			if err != nil {
+				return err
+			}
+			infrastructureConfigRawExtension.Raw = data
+		}
+		out.Spec.Provider.InfrastructureConfig = &garden.ProviderConfig{
+			RawExtension: infrastructureConfigRawExtension,
+		}
+
+		controlPlaneConfig := &awsv1alpha1.ControlPlaneConfig{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: awsv1alpha1.SchemeGroupVersion.String(),
+				Kind:       "ControlPlaneConfig",
+			},
+		}
+		var controlPlaneConfigOrigin string
+		if out.Spec.Provider.ControlPlaneConfig != nil {
```

Comment:
As discussed, this can never be true when converting from `v1beta1` -> `internal`.